lib/service/service.go

```diff
@@ -4311,9 +4312,31 @@ func (process *TeleportProcess) initProxyEndpoint(conn *Connector) error {
 	// from remote teleport nodes
 	var tsrv reversetunnelclient.Server
 	var peerClient *peer.Client
-
+	var peerQUICTransport *quic.Transport
 	if !process.Config.Proxy.DisableReverseTunnel {
 		if listeners.proxyPeer != nil {
+			// TODO(espadolini): allow this when the implementation is merged
+			if false && os.Getenv("TELEPORT_UNSTABLE_QUIC_PROXY_PEERING") == "yes" {
+				// the stateless reset key is important in case there's a crash
+				// so peers can be told to close their side of the connections
+				// instead of having to wait for a timeout; for this reason, we
+				// store it in the datadir, which should persist just as much as
+				// the host ID and the cluster credentials
+				resetKey, err := process.readOrInitPeerStatelessResetKey()
```

Comment:
For now it doesn't matter because it's only for proxies, but once we move the agent-to-agent communication to QUIC we must support storing it in the kubernetes secret

Reply:
This is specifically for proxy peering which is control plane only, and both in cloud and in the `teleport-cluster` chart the proxies are always a deployment with an emptyDir volume for the Teleport datadir, so this is clean enough for now; it's still not fully clear to me if we should have a fixed stateless reset key for client/agent to control plane at all, since it shouldn't be needed when using an ephemeral port to begin with.

Reply:
There's also a point to not _want_ to durably persist a secret with no expiration (as low-impact it might be); storing it in an emptyDir volume means that it won't ever moved between kube nodes - although if we decide we need a stable stateless reset key for agents we should consider mounting an emptyDir specifically for that even if the rest of the datadir is persisted (if somehow the agent is still using a PVC for the data dir).